Add support for Virtual Keyboard on Mobile Devices

acalatrava opened the pull request:
This should allow keyboard to appear on mobile devices. Tested on iOS. Issue #18

acalatrava committed:
Add support for Virtual Keyboard on Mobile Devices

jarrettgilliam commented:
Hi,

Nice find. This also works on android using chrome and firefox. It makes the desktop version look strange though. It has spell checking enabled and a blinking cursor.

![screen shot 2016-02-15 at 7 20 04 am](https://cloud.githubusercontent.com/assets/5099690/13049739/9d9d45e2-d3b4-11e5-9036-180299fefc46.png)

If you can get this to work without changing the appearance on the desktop, I'll merge it.

Thanks,
Jarrett

acalatrava committed:
Disable autocorrection on mobile and hack to hide blinking caret on desktop

Disabling autocorrection is straightforward, unfortunately the blinking
caret it’s not… I added a hack to hide the caret that will work on
desktop browsers but it won’t on mobile safari.

Caret is browser dependent and there is no CSS code to modify it.

acalatrava commented:
Disabling autocorrection was straightforward, unfortunately the blinking caret wasn't… I added a hack that will hide the caret on desktop browsers but it won’t on mobile safari.

Caret is browser dependent and there is no CSS code to modify it.

acalatrava committed:
Cosmetic fix

polycirclenet commented:
Was curious if this ever worked because my wetty doesn't work on chrome for android -- fairly late editions. I tried FF too and no dice.  Is it supposed to work?

foundry-admin commented:
We tested it for logging in to the term.
It works on iOS.
It does not work on Chrome phone (MotoX) or Chrome tablet.
Any workaround?

pketh commented:
Were Chrome phone (MotoX) or Chrome tablet working before/now? If not, can we get this merged? – partial support is better than no support

ghost commented:
Agreed, this should be merged. @krishnasrinivas

josefernandez91 commented:
@acalatrava Thanks for your modifications, now at least we have a keyboard. But I found some problems that impede the correct use, such as:
- When you write the text it looks bad.
- There is no scroll to visualize the story or if you list something very large you can not visualize it completely. The scroll does not exist either vertically or horizontally.

Can you correct those details to your solution?

regards

ghost commented:
This entire issue can be solved by updating to the latest version. https://transfer.sh/O1udA/hterm_all.js

josefernandez91 commented:
Thank you I'll try it and notice if it works. Currently use the commits that you made that are still not merged into the repository

josefernandez91 commented:
Great works perfect.
Only one comment to improve, every time you use the scroll up the keyboard is displayed, which is somewhat inconvenient to review content, it would be good if only the keyboard is displayed when pressed on the screen

etamponi commented:
Strange, I tried the same and it didn't work at all for me. Tested:

- Android Phone
- iPad Virtual Keyboard
- iPad Physical Keyboard

none of them work

ghost commented:
Did you see my post on support.glitch.com about this?

On Thu., Jun. 7, 2018 at 11:35 a.m. Emanuele Tamponi <
notifications@github.com> wrote:

> Strange, I tried the same and it didn't work at all for me. Tested:
>
>    - Android Phone
>    - iPad Virtual Keyboard
>    - iPad Physical Keyboard
>
> none of them work
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/krishnasrinivas/wetty/pull/49#issuecomment-395443832>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/Ahsw6p-VoFZIXgOyIMY8IZhFQoZ-QDiOks5t6TpOgaJpZM4HYjNv>
> .
>
-- 
Sent from a Minecrafters iPad

etamponi commented:
@nathfreder, yep, that's why I tried :)

nealey commented:
I would *much* rather have a blinking cursor on desktop, than have to maintain a fork of this so that I can restart downed services from my phone while I'm on vacation next week ;)

For what it's worth, I noticed a blinking cursor on desktop when using GCP ssh last week during a class. I said "huh, weird" and moved on with the class. It was fine.

Not being able to use wetty at all on a mobile device is a big problem for what I'm trying to do with it. Please just merge this and try to make it pretty later.
